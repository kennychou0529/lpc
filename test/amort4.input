16345.99
6.5
5
3
2012
