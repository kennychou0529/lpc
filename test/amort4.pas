                                     (* Chapter 16 - Program 4 *)
program AmortizationTable;

var Month : 1..12;
    StartingMonth : 1..12;
    Balance : real;
    Payment : real;
    InterestRate : real;
    AnnualAccumInterest : real;
    Year : integer;
    NumberOfYears : integer;
    OriginalLoan : real;


procedure CalculatePayment; (* **************** calculate payment *)
var Temp : real;
    Index : integer;
begin
   Temp := 1.0;
   for Index := 1 TO 12*NumberOfYears do
     Temp := Temp * (1.0 + InterestRate);
   Payment := OriginalLoan*InterestRate/(1.0 - 1.0/Temp);
end;

procedure InitializeData; (* ******************** initialize data *)
begin
   Writeln('   Pascal amortization program');
   Writeln;
   Write('Enter amount borrowed                         ');
   Readln(OriginalLoan);
   Balance := OriginalLoan;
   Write('Enter interest rate as percentage (i.e. 13.5) ');
   Readln(InterestRate);
   InterestRate := InterestRate/1200.0;
   Write('Enter number of years of payoff               ');
   Readln(NumberOfYears);
   Write('Enter month of first payment (i.e. 5 for May) ');
   Readln(StartingMonth);
   Write('Enter year of first payment (i.e. 1991)       ');
   Readln(Year);
   CalculatePayment;
   AnnualAccumInterest := 0.0; (* This is to accumulate Interest *)
end;

procedure PrintAnnualHeader; (* ************ print annual header *)
begin
   Writeln;
   Writeln;
   Writeln('Original loan amount = ',OriginalLoan:10:2,
           '   Interest rate = ',1200.0*InterestRate:6:2,'%');
   Writeln;
   Writeln('Month    payment  interest    princ   balance');
   Writeln;
end;

procedure CalculateAndPrint; (* ************ calculate and print *)
var InterestPayment : real;
    PrincipalPayment : real;
begin
   if Balance > 0.0 then begin
      InterestPayment := InterestRate * Balance;
      PrincipalPayment := Payment - InterestPayment;
      if PrincipalPayment > Balance then begin  (* loan payed off *)
         PrincipalPayment := Balance;              (* this month *)
         Payment := PrincipalPayment + InterestPayment;
         Balance := 0.0;
      end
      else begin  (* regular monthly payment *)
         Balance := Balance - PrincipalPayment;
      end;
      AnnualAccumInterest := AnnualAccumInterest
                                               + InterestPayment;
      Writeln(Month:5,Payment:10:2,InterestPayment:10:2,
              PrincipalPayment:10:2,Balance:10:2);
   end; (* of if Balance > 0.0 then *)
end;

procedure PrintAnnualSummary; (* ********** print annual summary *)
begin
   Writeln;
   Writeln('Total interest for ',Year:5,' = ',
            AnnualAccumInterest:10:2);
   AnnualAccumInterest := 0.0;
   Year := Year + 1;
   Writeln;
end;

begin   (* ******************************************* main program *)
   InitializeData;
   repeat
      PrintAnnualHeader;
      for Month := StartingMonth to 12 do begin
         CalculateAndPrint;
      end;
      PrintAnnualSummary;
      StartingMonth := 1;
   until Balance <= 0.0;
end. (* of main program *)
