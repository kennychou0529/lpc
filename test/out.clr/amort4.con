
compiling pass 1 (LPC)
-----------------------------------------------------
0 error(s), 0 warning(s)

P1 errcode = 0

compiling pass 2 (target = clr)
-----------------------------------------------------

P2 errcode = 0

verifying the resulting binary
-----------------------------------------------------
[IL]: Error: [amort4.exe : <Module>::InitializeData][offset 0x00000041][found address of Double][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [amort4.exe : <Module>::InitializeData][offset 0x00000071][found address of Double][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [amort4.exe : <Module>::InitializeData][offset 0x000000B5][found address of Int32][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [amort4.exe : <Module>::InitializeData][offset 0x000000DE][found address of Byte][expected unmanaged pointer] Unexpected type on the stack.
[IL]: Error: [amort4.exe : <Module>::InitializeData][offset 0x0000010E][found address of Int32][expected unmanaged pointer] Unexpected type on the stack.
5 Error(s) Verifying amort4.exe
