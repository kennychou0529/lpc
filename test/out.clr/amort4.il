
#include "commonDefs.il"

.field public static string[] _ProgramArgs

.method static void .cctor()
{
    ldc.i4 2
    newarr string
    dup
    ldc.i4 0
    ldstr "_input"
    stelem string
    dup
    ldc.i4 1
    ldstr "_output"
    stelem string
    stsfld string[] _ProgramArgs
    ret
}

// "amort4.pas"

// line 15
//================================================================================
// scope: CalculatePayment (level : 2)

.class @CalculatePayment
{
    // local variables
    .field public int32 Index
    .field public float64 Temp

    .method public void .ctor()
    {
        ldarg.0
        call instance void [mscorlib]System.Object::.ctor()
        ret
    }
}


// line 25
//================================================================================
// scope: InitializeData (level : 2)

.class @InitializeData
{
    .method public void .ctor()
    {
        ldarg.0
        call instance void [mscorlib]System.Object::.ctor()
        ret
    }
}


// line 45
//================================================================================
// scope: PrintAnnualHeader (level : 2)

.class @PrintAnnualHeader
{
    .method public void .ctor()
    {
        ldarg.0
        call instance void [mscorlib]System.Object::.ctor()
        ret
    }
}


// line 56
//================================================================================
// scope: CalculateAndPrint (level : 2)

.class @CalculateAndPrint
{
    // local variables
    .field public float64 InterestPayment
    .field public float64 PrincipalPayment

    .method public void .ctor()
    {
        ldarg.0
        call instance void [mscorlib]System.Object::.ctor()
        ret
    }
}


// line 78
//================================================================================
// scope: PrintAnnualSummary (level : 2)

.class @PrintAnnualSummary
{
    .method public void .ctor()
    {
        ldarg.0
        call instance void [mscorlib]System.Object::.ctor()
        ret
    }
}


// line 2
//================================================================================
// scope: program scope (level : 1)

// global variables
.class _G
{
    .field public static valuetype [lpcRuntime]LPC.File _input
    .field public static valuetype [lpcRuntime]LPC.File _output
    .field public static float64 AnnualAccumInterest
    .field public static float64 Balance
    .field public static float64 InterestRate
    .field public static int8 Month
    .field public static int32 NumberOfYears
    .field public static float64 OriginalLoan
    .field public static float64 Payment
    .field public static int8 StartingMonth
    .field public static int32 Year
}

// line 15

.method static
void CalculatePayment()
{
    .maxstack 32

    .locals init(
        class @CalculatePayment _F,
        int32 $tmpEndValue1,
        int32 $tmpIndex2,
        int32 $tmpIndex3)
    
    newobj instance void @CalculatePayment::.ctor()
    stloc _F

L_Entry:
    // line 19
    ldloc _F
    ldc.r8 1.00000
    stfld float64 @CalculatePayment::Temp
    // line 20
    ldc.i4 12
    ldsfld int32 _G::NumberOfYears
    mul
    stloc $tmpEndValue1
    ldc.i4 1
    dup
    ldloc $tmpEndValue1
    bgt L_for_end_2
    stloc $tmpIndex2
    ldloc _F
    ldloc $tmpIndex2
    stfld int32 @CalculatePayment::Index
L_for_loop_1:
    // line 21
    ldloc _F
    ldloc _F
    ldfld float64 @CalculatePayment::Temp
    ldc.r8 1.00000
    ldsfld float64 _G::InterestRate
    add
    mul
    stfld float64 @CalculatePayment::Temp
    ldloc _F
    ldfld int32 @CalculatePayment::Index
    dup
    ldloc $tmpEndValue1
    beq L_for_end_2
    ldc.i4.1
    add
    stloc $tmpIndex3
    ldloc _F
    ldloc $tmpIndex3
    stfld int32 @CalculatePayment::Index
    br L_for_loop_1
L_for_end_2:
    pop
    // line 22
    ldsfld float64 _G::OriginalLoan
    ldsfld float64 _G::InterestRate
    mul
    ldc.r8 1.00000
    ldc.r8 1.00000
    ldloc _F
    ldfld float64 @CalculatePayment::Temp
    div
    sub
    div
    stsfld float64 _G::Payment
    /* nop */

L_Epilogue:
    ret
}

// line 25

.method static
void InitializeData()
{
    .maxstack 32

    .locals init(class @InitializeData _F)
    
    newobj instance void @InitializeData::.ctor()
    stloc _F

L_Entry:
    // line 27
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    dup
    ldstr "   Pascal amortization program"
    call instance void [lpcRuntime]LPC.File::Write(string)
    call instance void [lpcRuntime]LPC.File::WriteLn()
    // line 28
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    call instance void [lpcRuntime]LPC.File::WriteLn()
    // line 29
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    dup
    ldstr "Enter amount borrowed                         "
    call instance void [lpcRuntime]LPC.File::Write(string)
    pop
    // line 30
    ldsflda valuetype [lpcRuntime]LPC.File _G::_input
    dup
    ldsflda float64 _G::OriginalLoan
    call instance void [lpcRuntime]LPC.File::Read(float64*)
    call instance void [lpcRuntime]LPC.File::ReadLn()
    // line 31
    ldsfld float64 _G::OriginalLoan
    stsfld float64 _G::Balance
    // line 32
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    dup
    ldstr "Enter interest rate as percentage (i.e. 13.5) "
    call instance void [lpcRuntime]LPC.File::Write(string)
    pop
    // line 33
    ldsflda valuetype [lpcRuntime]LPC.File _G::_input
    dup
    ldsflda float64 _G::InterestRate
    call instance void [lpcRuntime]LPC.File::Read(float64*)
    call instance void [lpcRuntime]LPC.File::ReadLn()
    // line 34
    ldsfld float64 _G::InterestRate
    ldc.r8 1200.00
    div
    stsfld float64 _G::InterestRate
    // line 35
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    dup
    ldstr "Enter number of years of payoff               "
    call instance void [lpcRuntime]LPC.File::Write(string)
    pop
    // line 36
    ldsflda valuetype [lpcRuntime]LPC.File _G::_input
    dup
    ldsflda int32 _G::NumberOfYears
    ldc.i4 -2147483648
    ldc.i4 2147483647
    call instance void [lpcRuntime]LPC.File::Read(int32*, int32, int32)
    call instance void [lpcRuntime]LPC.File::ReadLn()
    // line 37
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    dup
    ldstr "Enter month of first payment (i.e. 5 for May) "
    call instance void [lpcRuntime]LPC.File::Write(string)
    pop
    // line 38
    ldsflda valuetype [lpcRuntime]LPC.File _G::_input
    dup
    ldsflda int8 _G::StartingMonth
    ldc.i4 1
    ldc.i4 12
    call instance void [lpcRuntime]LPC.File::Read(int8*, int32, int32)
    call instance void [lpcRuntime]LPC.File::ReadLn()
    // line 39
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    dup
    ldstr "Enter year of first payment (i.e. 1991)       "
    call instance void [lpcRuntime]LPC.File::Write(string)
    pop
    // line 40
    ldsflda valuetype [lpcRuntime]LPC.File _G::_input
    dup
    ldsflda int32 _G::Year
    ldc.i4 -2147483648
    ldc.i4 2147483647
    call instance void [lpcRuntime]LPC.File::Read(int32*, int32, int32)
    call instance void [lpcRuntime]LPC.File::ReadLn()
    // line 41
    call void CalculatePayment()
    // line 42
    ldc.r8 0.000000
    stsfld float64 _G::AnnualAccumInterest
    /* nop */

L_Epilogue:
    ret
}

// line 45

.method static
void PrintAnnualHeader()
{
    .maxstack 32

    .locals init(class @PrintAnnualHeader _F)
    
    newobj instance void @PrintAnnualHeader::.ctor()
    stloc _F

L_Entry:
    // line 47
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    call instance void [lpcRuntime]LPC.File::WriteLn()
    // line 48
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    call instance void [lpcRuntime]LPC.File::WriteLn()
    // line 49
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    dup
    ldstr "Original loan amount = "
    call instance void [lpcRuntime]LPC.File::Write(string)
    dup
    ldc.i4 10
    ldc.i4 2
    ldsfld float64 _G::OriginalLoan
    call instance void [lpcRuntime]LPC.File::Format(int32, int32, float64)
    dup
    ldstr "   Interest rate = "
    call instance void [lpcRuntime]LPC.File::Write(string)
    dup
    ldc.i4 6
    ldc.i4 2
    ldc.r8 1200.00
    ldsfld float64 _G::InterestRate
    mul
    call instance void [lpcRuntime]LPC.File::Format(int32, int32, float64)
    dup
    ldc.i4 37 // '%'
    call instance void [lpcRuntime]LPC.File::Write(uint8)
    call instance void [lpcRuntime]LPC.File::WriteLn()
    // line 51
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    call instance void [lpcRuntime]LPC.File::WriteLn()
    // line 52
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    dup
    ldstr "Month    payment  interest    princ   balance"
    call instance void [lpcRuntime]LPC.File::Write(string)
    call instance void [lpcRuntime]LPC.File::WriteLn()
    // line 53
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    call instance void [lpcRuntime]LPC.File::WriteLn()
    /* nop */

L_Epilogue:
    ret
}

// line 56

.method static
void CalculateAndPrint()
{
    .maxstack 32

    .locals init(class @CalculateAndPrint _F)
    
    newobj instance void @CalculateAndPrint::.ctor()
    stloc _F

L_Entry:
    // line 60
    ldsfld float64 _G::Balance
    ldc.r8 0.000000
    cgt
    brfalse L_if_end_1
    // line 61
    ldloc _F
    ldsfld float64 _G::InterestRate
    ldsfld float64 _G::Balance
    mul
    stfld float64 @CalculateAndPrint::InterestPayment
    // line 62
    ldloc _F
    ldsfld float64 _G::Payment
    ldloc _F
    ldfld float64 @CalculateAndPrint::InterestPayment
    sub
    stfld float64 @CalculateAndPrint::PrincipalPayment
    // line 63
    ldloc _F
    ldfld float64 @CalculateAndPrint::PrincipalPayment
    ldsfld float64 _G::Balance
    cgt
    brfalse L_else_2
    // line 64
    ldloc _F
    ldsfld float64 _G::Balance
    stfld float64 @CalculateAndPrint::PrincipalPayment
    // line 65
    ldloc _F
    ldfld float64 @CalculateAndPrint::PrincipalPayment
    ldloc _F
    ldfld float64 @CalculateAndPrint::InterestPayment
    add
    stsfld float64 _G::Payment
    // line 66
    ldc.r8 0.000000
    stsfld float64 _G::Balance
    /* nop */
    br L_if_end_3
L_else_2:
    // line 69
    ldsfld float64 _G::Balance
    ldloc _F
    ldfld float64 @CalculateAndPrint::PrincipalPayment
    sub
    stsfld float64 _G::Balance
    /* nop */
L_if_end_3:
    // line 71
    ldsfld float64 _G::AnnualAccumInterest
    ldloc _F
    ldfld float64 @CalculateAndPrint::InterestPayment
    add
    stsfld float64 _G::AnnualAccumInterest
    // line 73
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    dup
    ldc.i4 5
    ldc.i4.0
    ldsfld int8 _G::Month
    call instance void [lpcRuntime]LPC.File::Format(int32, int32, int32)
    dup
    ldc.i4 10
    ldc.i4 2
    ldsfld float64 _G::Payment
    call instance void [lpcRuntime]LPC.File::Format(int32, int32, float64)
    dup
    ldc.i4 10
    ldc.i4 2
    ldloc _F
    ldfld float64 @CalculateAndPrint::InterestPayment
    call instance void [lpcRuntime]LPC.File::Format(int32, int32, float64)
    dup
    ldc.i4 10
    ldc.i4 2
    ldloc _F
    ldfld float64 @CalculateAndPrint::PrincipalPayment
    call instance void [lpcRuntime]LPC.File::Format(int32, int32, float64)
    dup
    ldc.i4 10
    ldc.i4 2
    ldsfld float64 _G::Balance
    call instance void [lpcRuntime]LPC.File::Format(int32, int32, float64)
    call instance void [lpcRuntime]LPC.File::WriteLn()
    /* nop */
L_if_end_1:
    /* nop */

L_Epilogue:
    ret
}

// line 78

.method static
void PrintAnnualSummary()
{
    .maxstack 32

    .locals init(class @PrintAnnualSummary _F)
    
    newobj instance void @PrintAnnualSummary::.ctor()
    stloc _F

L_Entry:
    // line 80
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    call instance void [lpcRuntime]LPC.File::WriteLn()
    // line 81
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    dup
    ldstr "Total interest for "
    call instance void [lpcRuntime]LPC.File::Write(string)
    dup
    ldc.i4 5
    ldc.i4.0
    ldsfld int32 _G::Year
    call instance void [lpcRuntime]LPC.File::Format(int32, int32, int32)
    dup
    ldstr " = "
    call instance void [lpcRuntime]LPC.File::Write(string)
    dup
    ldc.i4 10
    ldc.i4 2
    ldsfld float64 _G::AnnualAccumInterest
    call instance void [lpcRuntime]LPC.File::Format(int32, int32, float64)
    call instance void [lpcRuntime]LPC.File::WriteLn()
    // line 83
    ldc.r8 0.000000
    stsfld float64 _G::AnnualAccumInterest
    // line 84
    ldsfld int32 _G::Year
    ldc.i4 1
    add
    stsfld int32 _G::Year
    // line 85
    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    call instance void [lpcRuntime]LPC.File::WriteLn()
    /* nop */

L_Epilogue:
    ret
}

// line 0
//================================================================================
// scope: global scope (level : 0)

// line 2

.method static
void _Program()
{
    .maxstack 32

    .locals init(
        int8 $tmpEndValue1,
        int8 $tmpIndex2,
        int8 $tmpIndex3)

    ldsflda valuetype [lpcRuntime]LPC.File _G::_input
    ldc.i4 0
    call instance void [lpcRuntime]LPC.File::Init(int32)

    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    ldc.i4 1
    call instance void [lpcRuntime]LPC.File::Init(int32)

L_Entry:
    // line 89
    call void InitializeData()
    // line 90
L_repeat_1:
    // line 91
    call void PrintAnnualHeader()
    // line 92
    ldc.i4 12
    stloc $tmpEndValue1
    ldsfld int8 _G::StartingMonth
    dup
    ldloc $tmpEndValue1
    bgt L_for_end_3
    dup
    ldc.i4 1
    ldc.i4 12
    call void [lpcRuntime]LPC.Runtime::RTCheckRange(int32, int32, int32)
    stloc $tmpIndex2
    ldloc $tmpIndex2
    stsfld int8 _G::Month
L_for_loop_2:
    // line 93
    call void CalculateAndPrint()
    /* nop */
    ldsfld int8 _G::Month
    dup
    ldloc $tmpEndValue1
    beq L_for_end_3
    ldc.i4.1
    add
    dup
    ldc.i4 1
    ldc.i4 12
    call void [lpcRuntime]LPC.Runtime::RTCheckRange(int32, int32, int32)
    stloc $tmpIndex3
    ldloc $tmpIndex3
    stsfld int8 _G::Month
    br L_for_loop_2
L_for_end_3:
    pop
    // line 95
    call void PrintAnnualSummary()
    // line 96
    ldc.i4 1
    dup
    ldc.i4 1
    ldc.i4 12
    call void [lpcRuntime]LPC.Runtime::RTCheckRange(int32, int32, int32)
    stsfld int8 _G::StartingMonth
    /* nop */
    ldsfld float64 _G::Balance
    ldc.r8 0.000000
    cgt
    ldc.i4.0
    ceq
    brfalse L_repeat_1
    /* nop */
    leave L_Epilogue

L_Epilogue:
    ret

.try L_Entry to L_Epilogue finally
{
    ldsflda valuetype [lpcRuntime]LPC.File _G::_input
    call instance void [lpcRuntime]LPC.File::Close()

    ldsflda valuetype [lpcRuntime]LPC.File _G::_output
    call instance void [lpcRuntime]LPC.File::Close()

    endfinally
}

}
