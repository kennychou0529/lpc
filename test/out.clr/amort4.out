   Pascal amortization program

Enter amount borrowed                         Enter interest rate as percentage (i.e. 13.5) Enter number of years of payoff               Enter month of first payment (i.e. 5 for May) Enter year of first payment (i.e. 1991)       

Original loan amount =   16345.99   Interest rate =   6.50%

Month    payment  interest    princ   balance

    3    319.83     88.54    231.29  16114.70
    4    319.83     87.29    232.54  15882.16
    5    319.83     86.03    233.80  15648.36
    6    319.83     84.76    235.07  15413.30
    7    319.83     83.49    236.34  15176.96
    8    319.83     82.21    237.62  14939.34
    9    319.83     80.92    238.91  14700.43
   10    319.83     79.63    240.20  14460.23
   11    319.83     78.33    241.50  14218.73
   12    319.83     77.02    242.81  13975.92

Total interest for  2012 =     828.21



Original loan amount =   16345.99   Interest rate =   6.50%

Month    payment  interest    princ   balance

    1    319.83     75.70    244.13  13731.79
    2    319.83     74.38    245.45  13486.35
    3    319.83     73.05    246.78  13239.57
    4    319.83     71.71    248.11  12991.46
    5    319.83     70.37    249.46  12742.00
    6    319.83     69.02    250.81  12491.19
    7    319.83     67.66    252.17  12239.02
    8    319.83     66.29    253.53  11985.49
    9    319.83     64.92    254.91  11730.58
   10    319.83     63.54    256.29  11474.29
   11    319.83     62.15    257.68  11216.62
   12    319.83     60.76    259.07  10957.55

Total interest for  2013 =     819.56



Original loan amount =   16345.99   Interest rate =   6.50%

Month    payment  interest    princ   balance

    1    319.83     59.35    260.47  10697.07
    2    319.83     57.94    261.89  10435.19
    3    319.83     56.52    263.30  10171.88
    4    319.83     55.10    264.73   9907.15
    5    319.83     53.66    266.16   9640.99
    6    319.83     52.22    267.61   9373.38
    7    319.83     50.77    269.06   9104.33
    8    319.83     49.32    270.51   8833.81
    9    319.83     47.85    271.98   8561.83
   10    319.83     46.38    273.45   8288.38
   11    319.83     44.90    274.93   8013.45
   12    319.83     43.41    276.42   7737.03

Total interest for  2014 =     617.42



Original loan amount =   16345.99   Interest rate =   6.50%

Month    payment  interest    princ   balance

    1    319.83     41.91    277.92   7459.11
    2    319.83     40.40    279.42   7179.69
    3    319.83     38.89    280.94   6898.75
    4    319.83     37.37    282.46   6616.29
    5    319.83     35.84    283.99   6332.30
    6    319.83     34.30    285.53   6046.77
    7    319.83     32.75    287.07   5759.69
    8    319.83     31.20    288.63   5471.06
    9    319.83     29.63    290.19   5180.87
   10    319.83     28.06    291.77   4889.11
   11    319.83     26.48    293.35   4595.76
   12    319.83     24.89    294.93   4300.83

Total interest for  2015 =     401.73



Original loan amount =   16345.99   Interest rate =   6.50%

Month    payment  interest    princ   balance

    1    319.83     23.30    296.53   4004.30
    2    319.83     21.69    298.14   3706.16
    3    319.83     20.08    299.75   3406.40
    4    319.83     18.45    301.38   3105.03
    5    319.83     16.82    303.01   2802.02
    6    319.83     15.18    304.65   2497.37
    7    319.83     13.53    306.30   2191.07
    8    319.83     11.87    307.96   1883.11
    9    319.83     10.20    309.63   1573.48
   10    319.83      8.52    311.31   1262.17
   11    319.83      6.84    312.99    949.18
   12    319.83      5.14    314.69    634.50

Total interest for  2016 =     171.61



Original loan amount =   16345.99   Interest rate =   6.50%

Month    payment  interest    princ   balance

    1    319.83      3.44    316.39    318.10
    2    319.83      1.72    318.10      0.00

Total interest for  2017 =       5.16

